spring-cloud-dataflow-shell-core/src/main/java/org/springframework/cloud/dataflow/shell/autoconfigure/BaseShellAutoConfiguration.java

```diff
@@ -91,14 +92,33 @@ public void log() {
 	}
 
 	@Configuration
-	@ComponentScan({ "org.springframework.shell.commands", "org.springframework.cloud.dataflow.shell.command",
+	@ComponentScan({ "org.springframework.shell.commands", "org.springframework.cloud.dataflow.shell.command.common",
+			"org.springframework.cloud.dataflow.shell.command.classic",
 			"org.springframework.cloud.dataflow.shell.converter", "org.springframework.cloud.dataflow.shell.config" })
-	public static class RegisterInternalCommands {
+	@ConditionalOnProperty(name = "dataflow.mode", havingValue = "classic", matchIfMissing = true)
+	public static class ClassicRegisterInternalCommands {
```

Comment:
I understand this separation of packages gives a clear organization of classic/skipper commands which is nice. 

I am just wondering if we still need to support the case where some users could have already extended these commands and hence do we need to keep the existing commands (the to-be classic commands) in their current package itself. Also, I am sure this situation is not common and users would still be using our `DataFlowTemplate` for REST client extension, making it in the current package could be safer if we decide to support this case.

Reply:
Here is my opinion on this:
- In my understanding the shell commands are internal components rather a publicly supported API? 
- Furthermore even if we retain the common commands in the original o.s.c.d.s.command package we still will have to remove the existing AppRegistryCommand and StreamCommands from the common 'commands' package so we can control what gets loaded. 
- As pointed out this PR doesn't impact neither the REST APIs nor the XxxOptions templates